scripts/run_validation.py

```diff
@@ -173,13 +175,32 @@ def run_validation(args: namedtuple):
     datasets = get_datasets(data_service, data_path)
 
     start = time.time()
-    pool = Pool(args.pool_size)
-    results = pool.map(
-        partial(validate_single_rule, shared_cache, data_path, datasets, args),
-        rules,
-    )
-    pool.close()
-    pool.join()
+    results = []
+
+    # run each rule in a separate thread
+    with Pool(args.pool_size) as pool:
+        if args.disable_progressbar is True:
+            for rule_result in pool.imap_unordered(
```

Comment:
can we not just say results = pool.imap_unordered...

Reply:
We need to have it in a for loop, as we need to iterate the progress bar.

Reply:
I'm wrong, this is the part which avoids progressbar. Still imap_unordered returns an iterable rather than a list. I've tried it as `results = pool.imap_unordered...` and the process was just frozen at some point (although there were no issues).